test/generate_ssl_tests.pl

```diff
@@ -43,6 +43,7 @@ sub print_templates {
     # Add the implicit base configuration.
     foreach my $test (@ssltests::tests) {
         $test->{"server"} = { (%ssltests::base_server, %{$test->{"server"}}) };
+	$test->{"server2"} = { (%ssltests::base_server, %{$test->{"server2"}}) };
```

Comment:
Going with the "copy server config to server2 if not present" config approach. The .conf files will have an extra section, but the conf.in files won't, and it seems simpler.